### PR TITLE
Fixed comp

### PR DESCRIPTION
**Pull request by mlangow2**
Made changes to allow for cart_nstep and ang_step to scale during fixed_wiggle docking.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ dependencies:
  - python>=3.7
  - pyrosetta
  - numpy=1.16.3
+ - numba
  - matplotlib
  - pandas=0.24.2
```
Comment (akhmelin) [inline environment.yml:11]: can you add this in here? then I don't need to open a commit for it...
` - pyyaml`

File: ide/compile_file.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+   print()
+   os.system(cmd)
+
+   # cfg['include_dirs'] = ['../..', '../extern']
+   # cfg['compiler_args'] = ['-std=c++17', '-w', '-Ofast']
+   # cfg['dependencies'] = []
+   # cfg['parallel'] = False
+
+if __name__ == '__main__':
+   main()
```
Comment (akhmelin) [inline ide/compile_file.py:23]: sooo what is all this about? this is a hardcoded path to Will's directory, which is all fine for whomever is working in the IPD but will not work for the rest of the world..

File: rpxdock/app/options.py [hunk 3]
```diff
@@ -301,6 +309,15 @@ def default_cli_parser(parent=None, **kw):
    addarg("--ignored_aas", default='CGP', help='Amino acids to ignore in scoring. Default CGP')
    addarg("--score_self", action='store_true', default=False,
           help='score each interface seperately and dump in output pickle')
+   addarg("--recenter_input", action='store_true', default=False,
+          help='center coorinates from pdb input files')
```
Comment (akhmelin) [inline rpxdock/app/options.py:313]: typo in coordinates

is this for the very lazy people? :D

File: rpxdock/app/options.py [hunk 4]
```diff
@@ -301,6 +309,15 @@ def default_cli_parser(parent=None, **kw):
...
+
+  #addarg("--score_fun", default=standard,
+  #       help='apply this score function to rpx and ncontact')
+  #addarg("--weight_scorefun",
+  #       help='weights to use in score function')
+   addarg("--function", type=str, default='stnd',
+          help='score function to use for scoring')
```
Comment (akhmelin) [inline rpxdock/app/options.py:321]: line 319-320 is repetition of lines 322-323

File: rpxdock/app/pilot/helix_beta.py [hunk 5]
```diff
@@ -49,11 +49,26 @@ def get_helix_args():
...
-                       help='cartesian resolution of the initial search stage, default 10')
-   parser.add_argument("--angle_cell_width", type=float, default=30,
-                       help='angular resolution of the initial search stage, default 30')
+   # parser.add_argument("--cart_cell_width", type=float, default=10,
+   #                     help='cartesian resolution of the initial search stage, default 10')
+   # parser.add_argument("--angle_cell_width", type=float, default=30,
+   #                     help='angular resolution of the initial search stage, default 30')
+
+   parser.add_argument("--tether_xform", type=str, nargs='*', default=[],
+                       help='two structs with configuration to sample around')
```
Comment (akhmelin) [inline rpxdock/app/pilot/helix_beta.py:58]: be a little more descriptive please, from this help I have no idea what you are talking about (:

File: rpxdock/app/pilot/helix_beta.py [hunk 6]
```diff
@@ -49,11 +49,26 @@ def get_helix_args():
...
+   parser.add_argument("--tether_dist", type=float, default=3,
+                       help='max dist from supplied dimer configuration')
+   parser.add_argument("--tether_ang", type=float, default=10,
+                       help='max angle from supplied dimer configuration')
+
```
Comment (akhmelin) [inline rpxdock/app/pilot/helix_beta.py:63]: which dimer??

File: rpxdock/fragments/extract_fragments.py [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+      max_bb_redundancy=2.0,
+   )
+
+   # fname = '/home/sheffler/data/respairdat/pdb_res_pair_data_si30_100.pickle'
+   fname = '/home/sheffler/data/respairdat/pdb_res_pair_data_si30.pickle'
+   datfname = fname.replace('/home/sheffler/data/respairdat/pdb_res_pair_data', 'tmp_hlh_dat')
+   boundsfname = fname.replace('/home/sheffler/data/respairdat/pdb_res_pair_data',
+                               'tmp_hlh_bounds')
+   hlhfname = fname.replace('/home/sheffler/data/respairdat/pdb_res_pair_data', 'tmp_hlh')
+
```
Comment (akhmelin) [inline rpxdock/fragments/extract_fragments.py:38]: these dependancies will not work anywhere else either. Is there additional db that we should be sharing beyond the ones that are currently in the outwards folder?

File: rpxdock/search/onecomp.py [hunk 8]
```diff
@@ -147,6 +147,25 @@ def __call__(self, xforms, iresl=-1, wts={}, **kw):
...
+      '''
+      bounds: valid residue ranges to score after trimming i.e. don't score resi that were trimmed 
+      sfxn: hscore.scorepos scores stuff from the hscore that got passed 
+         takes two pos of bodies (the same monomer in this case)
+         xforms: not clashing xforms 
+         iresl: stage of hierarchical search (grid spacing: 4A --> 2A --> 1A --> 0.5A --> 0.25A)
+         sampling at highest resl probably 0.6A due to ori + cart
+         returns score # for each "dock"
+      '''
+
```
Comment (akhmelin) [inline rpxdock/search/onecomp.py:168]: how many times do we have this repeated?? let's clean up!

File: rpxdock/tests/search/test_cyclic.py [hunk 9]
```diff
@@ -65,9 +70,54 @@ def test_make_cyclic_hier_trim(hscore, body):
...
-   # test_make_cyclic_hier(hscore, body)
+
+   body = rp.Body('/home/sheffler/debug/marisa/input/dhr01.pdb', **kw)
+   result = rp.search.make_cyclic(body, "C2", hscore, **kw)
+   result.dump_pdbs_top_score(**kw.sub(nout_top=50, output_prefix='debug_marisa'))
+   print('num results:', len(result))
+   result = rp.concat_results([result])
+   rp.dump(result, 'result_test.pickle')
+
+   assert 0
```
Comment (akhmelin) [inline rpxdock/tests/search/test_cyclic.py:95]: not sure that the debug_marisa should be in the main branch

File: rpxdock/search/multicomp.py [hunk 10]
```diff
@@ -9,7 +9,7 @@ def make_multicomp(
    hscore,
    search=hier_search,
    sampler=None,
-   fixed_components=False,
+   components_already_aligned_to_sym_axes=False,
```
Comment (akhmelin) [inline rpxdock/search/multicomp.py:12]: I don't see this change in the options file?

File: rpxdock/tests/search/test_onecomp.py [hunk 11]
```diff
@@ -146,7 +147,7 @@ def test_cage_grid_onecomp_notrim(hscore, bodyC3):
    ref = rp.data.get_test_data('test_cage_grid_onecomp_notrim')
    rp.search.assert_results_close(result, ref)
 
-   # result.dump_pdbs_top_score(10)
+   # result.dump_pdbs_top_score(3)
    # assert 0
 
 def test_deepesh_1comp_bug(hscore):
```
Comment (akhmelin) [inline rpxdock/tests/search/test_onecomp.py:153]: would suggest deleting these specific debugging tests

**Review by akhmelin — CHANGES_REQUESTED** [review]
if this file is empty, can we delete it? `rpxdock/fragments/ball_joint.py`

**Comment by akhmelin** [comment]
this has way more changes then what is described in the commit...

**Comment by erincyang** [comment]
Mark, I think you have to pull master into your current branch and rerun pytest to resolve all of these comments Alena is pointing out.

**Comment by mlangow2** [comment]
I did pull master but didn't now I had to run a pytest, thanks for the heads up. Will get to it when I'm back tomorrow!

**Review by akhmelin — COMMENTED** [review]
I believe this did not change anything, at least I don't see any changes in the files..?

**Pull request closed without merging**